JoyboyCommunity/src/shared/components/Post.tsx

```diff
@@ -62,84 +168,111 @@ export const Post: React.FC<PostProps> = (props) => {
+      textOpacity.value = newLikes > 0 ? 1 : 0;
+      iconPosition.value = newLikes > 0 ? 0 : 6;
+      return newLikes;
+    });
+    scale.value = withSequence(
+      withTiming(1.5, {duration: 25, easing: Easing.out(Easing.ease)}),
+      withSpring(1, {damping: 6, stiffness: 200}), // Small rebound effect
+    );
+  };
+
+  useEffect(() => {
+    if (likes > 0) setIsLiked(true);
+  }, []);
+
   return (
     <PostCard>
+      {/* to show reposted message */}
       {repostedEvent && (
-        <View>
-          <Typography>Reposted</Typography>
-        </View>
+        <FlexView>
+          <RepostIcon color="#8f979d" height={18} />
+          <RepostMessage>Reposted</RepostMessage>
+        </FlexView>
       )}
       {/* TODO different rendering base on kind =1,6,7 and tags for kind = 1 */}
-      <PostLayout>
-        <View style={{marginRight: 10}}>
+      <PostHeader>
+        <FlexView>
           <Pressable onPress={() => handleProfilePress(event?.pubkey)}>
             <Image
               source={props?.sourceUser ?? require('../../../assets/joyboy-logo.png')}
-              style={{width: 44, height: 44}}
+              style={{width: 50, height: 50}}
             />
           </Pressable>
-        </View>
-
-        <View style={{gap: 4, flex: 1}}>
-          <Pressable onPress={handleNavigateToPostDetails}>
-            <Text style={{color: 'black', fontWeight: '700'}}>{event?.pubkey}</Text>
-
-            {post?.source && (
-              <Image
-                source={{uri: post.source}}
-                style={{
-                  width: '100%',
-
-                  height: 200,
-                  borderRadius: 8,
-                  marginTop: 8,
-                }}
-              />
-            )}
+
+          <Pressable onPress={handleNavigateToPostDetails} style={{marginLeft: 10}}>
+            {/* using static data for name */}
+            <PostTitle>Monkey D Luffy</PostTitle>
+            <PostSubtitleWrapper>
+              <PostSubtitle>@luffy</PostSubtitle>
+              <PostSubtitleCenterElement />
+              <PostSubtitle>{timestampToHumanReadable(event.created_at)}</PostSubtitle>
+            </PostSubtitleWrapper>
           </Pressable>
-        </View>
+        </FlexView>
 
-        {/* TODO check tags if it's:
+        <Pressable onPress={toggleLike}>
+          <LikeWrapper>
+            <AnimatedIcon style={animatedIconStyle}>
+              {isLiked ? <LikeFillIcon height={26} /> : <LikeIcon height={26} />}
+            </AnimatedIcon>
+            {likes > 0 && (
+              <LikeText style={animatedTextStyle}>
+                {likes} {likes === 1 ? 'like' : 'likes'}
+              </LikeText>
+            )}
+          </LikeWrapper>
+        </Pressable>
+      </PostHeader>
+
+      <PostContent>
+        <Pressable onPress={handleNavigateToPostDetails}>
+          <Text>{repostedEvent?.content ? repostedEvent?.content : event?.content}</Text>
```

Comment:
This can be shortened as `repostedEvent?.content ?? event?.content`

Reply:
ah, didnt catch my eye!